coreneuron/gpu/nrn_acc_manager.cpp

```diff
@@ -71,26 +71,33 @@ void cnrn_memcpy_to_device(void* d_ptr, void* h_ptr, size_t len) {
 #endif
 }
 
-void cnrn_target_delete(void* h_ptr, size_t len) {
+void* cnrn_target_deviceptr(void* h_ptr) {
 #if defined(CORENEURON_ENABLE_GPU) && !defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && defined(_OPENACC)
-    acc_delete(h_ptr, len);
+    return acc_deviceptr(h_ptr);
 #elif defined(CORENEURON_ENABLE_GPU) && defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && defined(_OPENMP)
-    (void)len;
+    void *d_ptr = nullptr;
     auto device_id = omp_get_default_device();
-    omp_target_disassociate_ptr(h_ptr, device_id);
-    auto* d_ptr = omp_get_mapped_ptr(h_ptr, device_id);
-    omp_target_free(d_ptr, device_id);
+
+    nrn_pragma_omp(target data device(device_id) use_device_ptr(h_ptr))
+    {
+        d_ptr = h_ptr;
+    }
+
+    return d_ptr;
 #else
-    throw std::runtime_error("cnrn_target_delete() not implemented without OpenACC/OpenMP and gpu build");
+    throw std::runtime_error("cnrn_target_deviceptr() not implemented without OpenACC/OpenMP and gpu build");
 #endif
 }
 
-void* cnrn_target_deviceptr(void* h_ptr) {
+void cnrn_target_delete(void* h_ptr, size_t len) {
 #if defined(CORENEURON_ENABLE_GPU) && !defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && defined(_OPENACC)
-    return acc_deviceptr(h_ptr);
+    acc_delete(h_ptr, len);
 #elif defined(CORENEURON_ENABLE_GPU) && defined(CORENEURON_PREFER_OPENMP_OFFLOAD) && defined(_OPENMP)
+    (void)len;
     auto device_id = omp_get_default_device();
-    return omp_get_mapped_ptr(h_ptr, device_id);
+    auto* d_ptr = cnrn_target_deviceptr(h_ptr);
+    omp_target_disassociate_ptr(h_ptr, device_id);
```

Comment:
This is not needed with https://github.com/BlueBrain/CoreNeuron/pull/708, right?